File: file/LevelFile.cpp
#include "LevelFile.h"

QPointF LevelFile::getStartPoint() const
{
    return _startPoint;
}

void LevelFile::setStartPoint(const QPointF &startPoint)
{
    _startPoint = startPoint;
}

int LevelFile::getFinishLine() const
{
    return _finishLine;
}

void LevelFile::setFinishLine(int finishLine)
{
    _finishLine = finishLine;
}

int LevelFile::getTimeToComplete() const
{
    return _timeToComplete;
}

void LevelFile::setTimeToComplete(int timeToComplete)
{
    _timeToComplete = timeToComplete;
}

QString LevelFile::getEnemyMapFile() const
{
    return _enemyMapFile;
}

void LevelFile::setEnemyMapFile(const QString &enemyMapFile)
{
    _enemyMapFile = enemyMapFile;
}

QString LevelFile::getTerrainFile() const
{
    return _terrainFile;
}

void LevelFile::setTerrainFile(const QString &terrainFile)
{
    _terrainFile = terrainFile;
}

QString LevelFile::getBackgroundLayerFile() const
{
    return _backgroundLayerFile;
}

void LevelFile::setBackgroundLayerFile(const QString &backgroundLayerFile)
{
    _backgroundLayerFile = backgroundLayerFile;
}

QString LevelFile::getBackForegroundLayerFile() const
{
    return _backForegroundLayerFile;
}

void LevelFile::setBackForegroundLayerFile(const QString &backForegroundLayerFile)
{
    _backForegroundLayerFile = backForegroundLayerFile;
}

QString LevelFile::getFunctionalLayerFile() const
{
    return _functionalLayerFile;
}

void LevelFile::setFunctionalLayerFile(const QString &functionalLayerFile)
{
    _functionalLayerFile = functionalLayerFile;
}

QString LevelFile::getForegroundLayerFile() const
{
    return _foregroundLayerFile;
}

void LevelFile::setForegroundLayerFile(const QString &foregroundLayerFile)
{
    _foregroundLayerFile = foregroundLayerFile;
}

File: external/FancyLand/lua/files/LevelFile.lua
 --- LevelFile ---
 
 -- Setup local access
 local LevelFile = {}
 local LevelFile_mt = {}
 LevelFile_mt.__index = LevelFile

 function LevelFile:new(startPoint,
                    finishLine,
                    timeToComplete,
                    enemyMapFile,
                    terrainFile,
                    backgroundLayerFile,
                    backForegroundLayerFile,
                    functionalLayerFile,
                    foregroundLayerFile)

    local levelFile = {}
    levelFile.startPoint = startPoint
    levelFile.finishLine = finishLine
    levelFile.timeToComplete = timeToComplete
    levelFile.enemyMapFile = enemyMapFile
    levelFile.terrainFile = terrainFile
    levelFile.backgroundLayerFile = backgroundLayerFile
    levelFile.backForegroundLayerFile = backForegroundLayerFile
    levelFile.functionalLayerFile = functionalLayerFile
    levelFile.foregroundLayerFile = foregroundLayerFile

    return setmetatable(levelFile, LevelFile_mt)
end

local function toString(f)
    return '[Level[StartPoint: ' .. tostring(f.startPoint) .. '][FinishLine: ' .. tostring(f.finishLine) ..
            '][TimeToComplete: ' .. tostring(f.timeToComplete) .. '][EnemyMapFile: ' .. f.enemyMapFile ..
            '][TerrainFile: ' .. f.terrainFile .. '][BackgroundLayerFile: ' .. f.backgroundLayerFile ..
            '][BackForegroundLayerFile: ' .. f.backForegroundLayerFile .. '][FunctionalLayerFile: ' ..
            f.functionalLayerFile .. '][ForegroundLayerFile: ' .. f.foregroundLayerFile .. ']]'
end

LevelFile_mt.__tostring = toString

return LevelFile
